src/EventStore.Core/ClusterVNode.cs

```diff
@@ -226,12 +228,18 @@ protected virtual void OnNodeStatusChanged(VNodeStatusChangeArgs e)
             _mainBus.Subscribe<SystemMessage.BecomeShutdown>(storageReader);
             monitoringRequestBus.Subscribe<MonitoringMessage.InternalStatsRequest>(storageReader);
 
-            var indexCommitterService = new IndexCommitterService(readIndex.IndexCommitter, _mainQueue, db.Config.ReplicationCheckpoint, db.Config.WriterCheckpoint, vNodeSettings.CommitAckCount);
+            // IO DISPATCHER
+            var ioDispatcher = new IODispatcher(_mainQueue, new PublishEnvelope(_mainQueue));
+            var scavengerLogManager = new TFChunkScavengerLogManager(_nodeInfo.ExternalHttp.ToString(), TimeSpan.FromDays(vNodeSettings.ScavengeHistoryMaxAge), ioDispatcher);
+
+            var indexCommitterService = new IndexCommitterService(readIndex.IndexCommitter, _mainQueue,
+                db.Config.ReplicationCheckpoint, db.Config.WriterCheckpoint, vNodeSettings.CommitAckCount, tableIndex, scavengerLogManager);
```

Comment:
These changes need to be reverted. See: https://github.com/EventStore/EventStore/pull/1786#pullrequestreview-189685299